crates/cxx-qt-gen/src/generator/rust/invokable.rs

```diff
@@ -50,46 +45,29 @@ pub fn generate_rust_invokables(
                     quote! { #ident: #ty }
                 })
                 .collect::<Vec<TokenStream>>();
-            quote! { self: #rust_struct, cpp: #cpp_struct, #(#parameters),* }
+            quote! { self: #cpp_struct, #(#parameters),* }
         };
         let return_type = &invokable.method.sig.output;
-        let has_return = if matches!(invokable.method.sig.output, ReturnType::Default) {
-            quote! {}
-        } else {
-            quote! { return }
-        };
 
         let mut unsafe_block = None;
         let mut unsafe_call = Some(quote! { unsafe });
         if invokable.safe {
             std::mem::swap(&mut unsafe_call, &mut unsafe_block);
         }
 
-        let parameter_names = invokable
-            .parameters
-            .iter()
-            .map(|parameter| parameter.ident.clone())
-            .collect::<Vec<Ident>>();
-
         let fragment = RustFragmentPair {
             cxx_bridge: vec![quote! {
-                // TODO: is an unsafe block valid?
+                // Note: extern "Rust" block does not need to be unsafe
                 extern "Rust" {
```

Comment:
we should confirm that `extern "Rust"` with an opaque C++ type and it's behaviour we are using here is valid upstream and ensure there is a test.